### PR TITLE
Add models for referral system (referral, marketing) and additional test coverage

### PR DESCRIPTION
**Pull request by barbaraliau**
More or less done and ready for review; but not ready to merge until other components are ready.

Also includes some refactoring for email validation --> pulled into a `utils.js` file since multiple models now need to use it.

### CODE REVIEW

**Comment by barbaraliau** [comment]
This should be ready for review when you guys get a chance @dylanlott @braydonf

File: index.js [hunk 1]
```diff
@@ -5,10 +5,9 @@ const mongoose = require('mongoose');
 const merge = require('merge');
 require('dotenv').config({ silent: true });
 
-require('mongoose-currency').loadType(mongoose);
 require('mongoose-types').loadTypes(mongoose);
 
-mongoose.Promise = global.Promise;
+mongoose.Promise = require('bluebird');
```
Comment (braydonf) [inline index.js:10]: Do we need to use a library for this, I'd prefer to use what's built-in.

File: .travis.yml [hunk 2]
```diff
@@ -11,6 +15,15 @@ addons:
     packages:
       - mongodb-org-server
       - mongodb-org-shell
+
+cache:
```
Comment (braydonf) [inline .travis.yml]: These changes in travis.yml I don't think are necessary anymore.

Comment (barbaraliau) [reply .travis.yml]: Oh yes. I forgot these got pulled in from a previous branch.

**Comment by braydonf** [comment]
This looks like it's mostly around debit and credit, so review from @bryanchriswhite on that would be useful as well as @pgerbes1

File: index.js (hunk 1 above)
Comment (barbaraliau) [reply index.js:10]: We're using a lot of promises in regards to billing and referral models, and bluebird is more performant and handles errors better.

**Review by bryanchriswhite — APPROVED** [review]
@braydonf you  wanna have a look... can we merge, tag, publish, etc.?

**Review by dylanlott — APPROVED** [review]
Went thru this with @bryanchriswhite and everything looks good to go. Need these changes published ASAP so Referrals can go to production

**Pull request merged**